src/ploigos_step_runner/results/workflow_result.py

```diff
@@ -239,8 +234,49 @@ def load_from_pickle_file(pickle_filename):
         except Exception as error:
             raise StepRunnerException(f'error loading {pickle_filename}: {error}') from error
 
+    def merge_with_pickle_file(self, pickle_filename):
+        """Merge our workflow list with that stored on disk.
+        When we find overlaps, our in-memory values win.
+
+        Parameters
+        ----------
+        pickle_filename : the on-disk path to the pickle file to merge with.
+
+        """
+        on_disk_wfr = WorkflowResult.load_from_pickle_file(pickle_filename)
+        on_disk_results = on_disk_wfr.workflow_list
+        merged_workflow_list = []
+
+        for in_mem_step_result in self.__workflow_list:
+            on_disk_step_result = on_disk_wfr.get_step_result(
+                                    step_name=in_mem_step_result.step_name,
```

Comment:
nit: indentation here is not consistent with the rest of src and how we do multi line indentiation.